[quirc, opencv4] Build fixes from 2021-01-18 build

BillyONeal opened the pull request:
Build fixes from 2020-01-18 build https://dev.azure.com/vcpkg/public/_build/results?buildId=48065

> PASSING, REMOVE FROM FAIL LIST: chartdir:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: cmark:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: cppcms:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: cudnn:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: devicenameresolver:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: dxut:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: epsilon:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: fann:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: getopt-win32:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: glib:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: graphqlparser:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: halide:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: hpx:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: jemalloc:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: libepoxy:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: libhsplasma:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: libmodman:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: libpcap:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: libplist:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: milerius-sfml-imgui:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: monkeys-audio:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: moos-core:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: msix:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: ngspice:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: openni2:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: openvr:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: pmdk:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: pngpp:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: qpid-proton:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: qt5-tools:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: sciter:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: scylla-wrapper:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: soundtouch:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: spirv-tools:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: torch-th:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: treehopper:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: unrar:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: urdfdom:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: xalan-c:x64-windows-static-md (.\scripts\ci.baseline.txt)

These look like they have always passed since we turned on x64-windows-static-md in nightly CI.

>PASSING, REMOVE FROM FAIL LIST: osgearth:x64-windows (.\scripts\ci.baseline.txt)
>PASSING, REMOVE FROM FAIL LIST: osgearth:x86-windows (.\scripts\ci.baseline.txt)

Maybe addressed by https://github.com/microsoft/vcpkg/pull/15777

>REGRESSION: gettext:arm-uwp. If expected, add gettext:arm-uwp=fail to .\scripts\ci.baseline.txt.
>REGRESSION: gettext:arm64-windows. If expected, add gettext:arm64-windows=fail to .\scripts\ci.baseline.txt.

Probably fixed by https://github.com/microsoft/vcpkg/pull/15740

> PASSING, REMOVE FROM FAIL LIST: directxmesh:x64-linux (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: uvatlas:x64-linux (.\scripts\ci.baseline.txt)

Probably an error in the authoring of https://github.com/microsoft/vcpkg/pull/15554/

> REGRESSION: dcmtk:x86-windows. If expected, add dcmtk:x86-windows=fail to .\scripts\ci.baseline.txt.
>
> "C:\PROGRA~2\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29333\bin\Hostx64\x86\cl.exe"   /TP -DDCMTK_BUILD_IN_PROGRESS -DNOMINMAX -DUSE_NULL_SAFE_OFSTRING -D_CRT_FAR_MAPPINGS_NO_DEPRECATE -D_CRT_IS_WCTYPE_NO_DEPRECATE -D_CRT_MANAGED_FP_NO_DEPRECATE -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_SECURE_NO_DEPRECATE -D_CRT_SECURE_NO_DEPRECATE_GLOBALS -D_CRT_SETERRORMODE_BEEP_SLEEP_NO_DEPRECATE -D_CRT_TIME_FUNCTIONS_NO_DEPRECATE -D_CRT_VCCLRIT_NO_DEPRECATE -D_REENTRANT -D_SCL_SECURE_NO_DEPRECATE -Dofstd_EXPORTS -ID:\installed\x86-windows\include -Iconfig\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\ofstd\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\oflog\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmdata\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmimgle\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmimage\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmjpeg\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmjpls\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmtls\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmnet\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmsr\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmsign\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmwlm\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmqrdb\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmpstat\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmrt\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmiod\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmfg\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmseg\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmtract\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmpmap\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmect\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -DDEBUG /showIncludes /Foofstd\libsrc\CMakeFiles\ofstd.dir\offilsys.cc.obj /Fdofstd\libsrc\CMakeFiles\ofstd.dir\ofstd.pdb /FS -c D:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\ofstd\libsrc\offilsys.cc
> D:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\ofstd\libsrc\offilsys.cc(376): error C2039: 'DIR': is not a member of '`global namespace''
>
> REGRESSION: tesseract:x64-windows-static-md. If expected, add tesseract:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.
> REGRESSION: tesseract:x64-windows-static. If expected, add tesseract:x64-windows-static=fail to .\scripts\ci.baseline.txt.
> REGRESSION: tesseract:x64-windows. If expected, add tesseract:x64-windows=fail to .\scripts\ci.baseline.txt.
> REGRESSION: tesseract:x86-windows. If expected, add tesseract:x86-windows=fail to .\scripts\ci.baseline.txt.
> [2/241] "C:\PROGRA~2\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29333\bin\Hostx64\x86\cl.exe"   /TP -DHAVE_CONFIG_H -DTESS_EXPORTS -D_CRT_SECURE_NO_WARNINGS -D_SILENCE_STDEXT_HASH_DEPRECATION_WARNINGS=1 -D__AVX2__ -D__AVX__ -D__BMI2__ -D__FMA__ -D__SSE2__ -D__SSE3__ -D__SSE4A__ -D__SSE4_1__ -D__SSE4_2__ -D__SSSE3__ -Dlibtesseract_EXPORTS -ID:\installed\x86-windows\share\leptonica\..\..\include -ID:\installed\x86-windows\share\leptonica\..\..\include\leptonica -I. -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\api -Iapi -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\arch -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\ccmain -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\ccstruct -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\ccutil -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\classify -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\cutil -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\dict -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\lstm -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\opencl -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\textord -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\viewer -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\wordrec -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\training -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\vs2010\tesseract -ID:\installed\x86-windows\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /utf-8 /MP /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  /W4 /bigobj -std:c++17 /showIncludes /FoCMakeFiles\libtesseract.dir\src\classify\shapetable.cpp.obj /FdCMakeFiles\libtesseract.dir\ /FS -c D:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\classify\shapetable.cpp
> FAILED: CMakeFiles/libtesseract.dir/src/classify/shapetable.cpp.obj
> "C:\PROGRA~2\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29333\bin\Hostx64\x86\cl.exe"   /TP -DHAVE_CONFIG_H -DTESS_EXPORTS -D_CRT_SECURE_NO_WARNINGS -D_SILENCE_STDEXT_HASH_DEPRECATION_WARNINGS=1 -D__AVX2__ -D__AVX__ -D__BMI2__ -D__FMA__ -D__SSE2__ -D__SSE3__ -D__SSE4A__ -D__SSE4_1__ -D__SSE4_2__ -D__SSSE3__ -Dlibtesseract_EXPORTS -ID:\installed\x86-windows\share\leptonica\..\..\include -ID:\installed\x86-windows\share\leptonica\..\..\include\leptonica -I. -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\api -Iapi -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\arch -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\ccmain -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\ccstruct -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\ccutil -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\classify -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\cutil -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\dict -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\lstm -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\opencl -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\textord -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\viewer -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\wordrec -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\training -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\vs2010\tesseract -ID:\installed\x86-windows\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /utf-8 /MP /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  /W4 /bigobj -std:c++17 /showIncludes /FoCMakeFiles\libtesseract.dir\src\classify\shapetable.cpp.obj /FdCMakeFiles\libtesseract.dir\ /FS -c D:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\classify\shapetable.cpp
> cl : Command line warning D9025 : overriding '/W3' with '/W4'
> D:\installed\x86-windows\include\params.h(36): error C2143: syntax error: missing ';' before '*'

https://github.com/microsoft/vcpkg/pull/15785

> REGRESSION: fluidsynth:x64-windows-static-md. If expected, add fluidsynth:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.

Was previously a cascaded failure from glib.

REGRESSION: libevent:x86-windows. If expected, add libevent:x86-windows=fail to .\scripts\ci.baseline.txt.

> REGRESSION: ideviceinstaller:x64-windows-static-md. If expected, add ideviceinstaller:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.
> REGRESSION: libcrafter:x64-windows-static-md. If expected, add libcrafter:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.
> REGRESSION: libirecovery:x64-windows-static-md. If expected, add libirecovery:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.
> REGRESSION: moos-essential:x64-windows-static-md. If expected, add moos-essential:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.
> REGRESSION: openmama:x64-windows-static-md. If expected, add openmama:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.
> REGRESSION: sdformat10:x64-windows-static-md. If expected, add sdformat10:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.
> REGRESSION: sdformat6:x64-windows-static-md. If expected, add sdformat6:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.
> REGRESSION: sdformat9:x64-windows-static-md. If expected, add sdformat9:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.

I'm going to assume these were cascaded failures for now.

> REGRESSION: mesa:x64-windows-static-md. If expected, add mesa:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.
> REGRESSION: mesa:x64-windows. If expected, add mesa:x64-windows=fail to .\scripts\ci.baseline.txt.
> REGRESSION: mesa:x86-windows. If expected, add mesa:x86-windows=fail to .\scripts\ci.baseline.txt.

Nancy is working on this https://github.com/microsoft/vcpkg/pull/15729

> REGRESSION: quirc:x64-linux. If expected, add quirc:x64-linux=fail to .\scripts\ci.baseline.txt.
> REGRESSION: quirc:x64-osx. If expected, add quirc:x64-osx=fail to .\scripts\ci.baseline.txt.
> Starting package 1166/1442: quirc:x64-linux
> Building package quirc[core]:x64-linux...
> -- Downloading https://github.com/dlbeer/quirc/archive/7e7ab596e4d0988faf1c12ae89c354b114181c40.tar.gz -> dlbeer-quirc-7e7ab596e4d0988faf1c12ae89c354b114181c40.tar.gz...
> -- Extracting source /mnt/vcpkg-ci/downloads/dlbeer-quirc-7e7ab596e4d0988faf1c12ae89c354b114181c40.tar.gz
> -- Using source at /mnt/vcpkg-ci/buildtrees/quirc/src/b114181c40-f1e71e0d5b.clean
> -- Configuring x64-linux-dbg
> -- Configuring x64-linux-rel
> -- Building x64-linux-dbg
> -- Building x64-linux-rel
> -- Installing: /mnt/vcpkg-ci/packages/quirc_x64-linux/share/quirc/copyright/LICENSE
> -- Performing post-build validation
> -- Performing post-build validation done
> Uploaded binaries to 1 HTTP remotes.
> Building package quirc[core]:x64-linux... done
> Installing package quirc[core]:x64-linux...
> The following files are already installed in /mnt/vcpkg-ci/installed/x64-linux and are in conflict with quirc:x64-linux
>
> Installed by opencv4:x64-linux
>     lib/libquirc.a
>
> Elapsed time for package quirc:x64-linux: 2.531 s

Probably broken by https://github.com/microsoft/vcpkg/pull/15089, fix here

> REGRESSION: spatialite-tools:x64-linux. If expected, add spatialite-tools:x64-linux=fail to .\scripts\ci.baseline.txt.
> REGRESSION: spatialite-tools:x64-osx. If expected, add spatialite-tools:x64-osx=fail to .\scripts\ci.baseline.txt.

These are spewing linker errors, probably triggered by https://github.com/microsoft/vcpkg/pull/15768/ although that PR removed them from ci.baseline.txt?

BillyONeal committed:
[quirc, opencv4] Use a cmake package for quirc and devendor quirc from opencv4.

BillyONeal committed:
Build fixes from 2020-01-18 build

https://dev.azure.com/vcpkg/public/_build/results?buildId=48065

> PASSING, REMOVE FROM FAIL LIST: chartdir:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: cmark:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: cppcms:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: cudnn:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: devicenameresolver:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: dxut:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: epsilon:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: fann:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: getopt-win32:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: glib:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: graphqlparser:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: halide:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: hpx:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: jemalloc:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: libepoxy:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: libhsplasma:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: libmodman:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: libpcap:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: libplist:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: milerius-sfml-imgui:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: monkeys-audio:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: moos-core:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: msix:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: ngspice:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: openni2:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: openvr:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: pmdk:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: pngpp:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: qpid-proton:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: qt5-tools:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: sciter:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: scylla-wrapper:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: soundtouch:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: spirv-tools:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: torch-th:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: treehopper:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: unrar:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: urdfdom:x64-windows-static-md (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: xalan-c:x64-windows-static-md (.\scripts\ci.baseline.txt)

These look like they have always passed since we turned on x64-windows-static-md in nightly CI.

>PASSING, REMOVE FROM FAIL LIST: osgearth:x64-windows (.\scripts\ci.baseline.txt)
>PASSING, REMOVE FROM FAIL LIST: osgearth:x86-windows (.\scripts\ci.baseline.txt)

Maybe addressed by https://github.com/microsoft/vcpkg/pull/15777

>REGRESSION: gettext:arm-uwp. If expected, add gettext:arm-uwp=fail to .\scripts\ci.baseline.txt.
>REGRESSION: gettext:arm64-windows. If expected, add gettext:arm64-windows=fail to .\scripts\ci.baseline.txt.

Probably fixed by https://github.com/microsoft/vcpkg/pull/15740

> PASSING, REMOVE FROM FAIL LIST: directxmesh:x64-linux (.\scripts\ci.baseline.txt)
> PASSING, REMOVE FROM FAIL LIST: uvatlas:x64-linux (.\scripts\ci.baseline.txt)

Probably an error in the authoring of https://github.com/microsoft/vcpkg/pull/15554/

> REGRESSION: dcmtk:x86-windows. If expected, add dcmtk:x86-windows=fail to .\scripts\ci.baseline.txt.
>
> "C:\PROGRA~2\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29333\bin\Hostx64\x86\cl.exe"   /TP -DDCMTK_BUILD_IN_PROGRESS -DNOMINMAX -DUSE_NULL_SAFE_OFSTRING -D_CRT_FAR_MAPPINGS_NO_DEPRECATE -D_CRT_IS_WCTYPE_NO_DEPRECATE -D_CRT_MANAGED_FP_NO_DEPRECATE -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_SECURE_NO_DEPRECATE -D_CRT_SECURE_NO_DEPRECATE_GLOBALS -D_CRT_SETERRORMODE_BEEP_SLEEP_NO_DEPRECATE -D_CRT_TIME_FUNCTIONS_NO_DEPRECATE -D_CRT_VCCLRIT_NO_DEPRECATE -D_REENTRANT -D_SCL_SECURE_NO_DEPRECATE -Dofstd_EXPORTS -ID:\installed\x86-windows\include -Iconfig\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\ofstd\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\oflog\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmdata\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmimgle\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmimage\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmjpeg\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmjpls\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmtls\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmnet\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmsr\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmsign\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmwlm\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmqrdb\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmpstat\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmrt\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmiod\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmfg\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmseg\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmtract\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmpmap\include -ID:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\dcmect\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -DDEBUG /showIncludes /Foofstd\libsrc\CMakeFiles\ofstd.dir\offilsys.cc.obj /Fdofstd\libsrc\CMakeFiles\ofstd.dir\ofstd.pdb /FS -c D:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\ofstd\libsrc\offilsys.cc
> D:\buildtrees\dcmtk\src\95da808b49-b82aa35f79.clean\ofstd\libsrc\offilsys.cc(376): error C2039: 'DIR': is not a member of '`global namespace''
>
> REGRESSION: tesseract:x64-windows-static-md. If expected, add tesseract:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.
> REGRESSION: tesseract:x64-windows-static. If expected, add tesseract:x64-windows-static=fail to .\scripts\ci.baseline.txt.
> REGRESSION: tesseract:x64-windows. If expected, add tesseract:x64-windows=fail to .\scripts\ci.baseline.txt.
> REGRESSION: tesseract:x86-windows. If expected, add tesseract:x86-windows=fail to .\scripts\ci.baseline.txt.
> [2/241] "C:\PROGRA~2\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29333\bin\Hostx64\x86\cl.exe"   /TP -DHAVE_CONFIG_H -DTESS_EXPORTS -D_CRT_SECURE_NO_WARNINGS -D_SILENCE_STDEXT_HASH_DEPRECATION_WARNINGS=1 -D__AVX2__ -D__AVX__ -D__BMI2__ -D__FMA__ -D__SSE2__ -D__SSE3__ -D__SSE4A__ -D__SSE4_1__ -D__SSE4_2__ -D__SSSE3__ -Dlibtesseract_EXPORTS -ID:\installed\x86-windows\share\leptonica\..\..\include -ID:\installed\x86-windows\share\leptonica\..\..\include\leptonica -I. -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\api -Iapi -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\arch -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\ccmain -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\ccstruct -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\ccutil -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\classify -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\cutil -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\dict -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\lstm -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\opencl -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\textord -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\viewer -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\wordrec -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\training -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\vs2010\tesseract -ID:\installed\x86-windows\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /utf-8 /MP /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  /W4 /bigobj -std:c++17 /showIncludes /FoCMakeFiles\libtesseract.dir\src\classify\shapetable.cpp.obj /FdCMakeFiles\libtesseract.dir\ /FS -c D:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\classify\shapetable.cpp
> FAILED: CMakeFiles/libtesseract.dir/src/classify/shapetable.cpp.obj
> "C:\PROGRA~2\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29333\bin\Hostx64\x86\cl.exe"   /TP -DHAVE_CONFIG_H -DTESS_EXPORTS -D_CRT_SECURE_NO_WARNINGS -D_SILENCE_STDEXT_HASH_DEPRECATION_WARNINGS=1 -D__AVX2__ -D__AVX__ -D__BMI2__ -D__FMA__ -D__SSE2__ -D__SSE3__ -D__SSE4A__ -D__SSE4_1__ -D__SSE4_2__ -D__SSSE3__ -Dlibtesseract_EXPORTS -ID:\installed\x86-windows\share\leptonica\..\..\include -ID:\installed\x86-windows\share\leptonica\..\..\include\leptonica -I. -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\api -Iapi -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\arch -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\ccmain -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\ccstruct -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\ccutil -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\classify -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\cutil -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\dict -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\lstm -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\opencl -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\textord -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\viewer -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\wordrec -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\training -ID:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\vs2010\tesseract -ID:\installed\x86-windows\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /utf-8 /MP /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  /W4 /bigobj -std:c++17 /showIncludes /FoCMakeFiles\libtesseract.dir\src\classify\shapetable.cpp.obj /FdCMakeFiles\libtesseract.dir\ /FS -c D:\buildtrees\tesseract\src\4.1.1-3a7d5a1d2b.clean\src\classify\shapetable.cpp
> cl : Command line warning D9025 : overriding '/W3' with '/W4'
> D:\installed\x86-windows\include\params.h(36): error C2143: syntax error: missing ';' before '*'

https://github.com/microsoft/vcpkg/pull/15785

> REGRESSION: fluidsynth:x64-windows-static-md. If expected, add fluidsynth:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.

Was previously a cascaded failure from glib.

REGRESSION: libevent:x86-windows. If expected, add libevent:x86-windows=fail to .\scripts\ci.baseline.txt.

> REGRESSION: ideviceinstaller:x64-windows-static-md. If expected, add ideviceinstaller:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.
> REGRESSION: libcrafter:x64-windows-static-md. If expected, add libcrafter:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.
> REGRESSION: libirecovery:x64-windows-static-md. If expected, add libirecovery:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.
> REGRESSION: moos-essential:x64-windows-static-md. If expected, add moos-essential:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.
> REGRESSION: openmama:x64-windows-static-md. If expected, add openmama:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.
> REGRESSION: sdformat10:x64-windows-static-md. If expected, add sdformat10:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.
> REGRESSION: sdformat6:x64-windows-static-md. If expected, add sdformat6:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.
> REGRESSION: sdformat9:x64-windows-static-md. If expected, add sdformat9:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.

I'm going to assume these were cascaded failures for now.

> REGRESSION: mesa:x64-windows-static-md. If expected, add mesa:x64-windows-static-md=fail to .\scripts\ci.baseline.txt.
> REGRESSION: mesa:x64-windows. If expected, add mesa:x64-windows=fail to .\scripts\ci.baseline.txt.
> REGRESSION: mesa:x86-windows. If expected, add mesa:x86-windows=fail to .\scripts\ci.baseline.txt.

Nancy is working on this https://github.com/microsoft/vcpkg/pull/15729

> REGRESSION: quirc:x64-linux. If expected, add quirc:x64-linux=fail to .\scripts\ci.baseline.txt.
> REGRESSION: quirc:x64-osx. If expected, add quirc:x64-osx=fail to .\scripts\ci.baseline.txt.
> Starting package 1166/1442: quirc:x64-linux
> Building package quirc[core]:x64-linux...
> -- Downloading https://github.com/dlbeer/quirc/archive/7e7ab596e4d0988faf1c12ae89c354b114181c40.tar.gz -> dlbeer-quirc-7e7ab596e4d0988faf1c12ae89c354b114181c40.tar.gz...
> -- Extracting source /mnt/vcpkg-ci/downloads/dlbeer-quirc-7e7ab596e4d0988faf1c12ae89c354b114181c40.tar.gz
> -- Using source at /mnt/vcpkg-ci/buildtrees/quirc/src/b114181c40-f1e71e0d5b.clean
> -- Configuring x64-linux-dbg
> -- Configuring x64-linux-rel
> -- Building x64-linux-dbg
> -- Building x64-linux-rel
> -- Installing: /mnt/vcpkg-ci/packages/quirc_x64-linux/share/quirc/copyright/LICENSE
> -- Performing post-build validation
> -- Performing post-build validation done
> Uploaded binaries to 1 HTTP remotes.
> Building package quirc[core]:x64-linux... done
> Installing package quirc[core]:x64-linux...
> The following files are already installed in /mnt/vcpkg-ci/installed/x64-linux and are in conflict with quirc:x64-linux
>
> Installed by opencv4:x64-linux
>     lib/libquirc.a
>
> Elapsed time for package quirc:x64-linux: 2.531 s

Probably broken by https://github.com/microsoft/vcpkg/pull/15089, fix here

> REGRESSION: spatialite-tools:x64-linux. If expected, add spatialite-tools:x64-linux=fail to .\scripts\ci.baseline.txt.
> REGRESSION: spatialite-tools:x64-osx. If expected, add spatialite-tools:x64-osx=fail to .\scripts\ci.baseline.txt.

These are spewing linker errors, probably triggered by https://github.com/microsoft/vcpkg/pull/15768/ although that PR removed them from ci.baseline.txt?

BillyONeal committed:
Fixup spurious / and format-manifest.

BillyONeal committed:
x-add-version

BillyONeal commented:
I'm still very new at this cmake config stuff so @ras0219 @strega-nil @vicroms @dan-shaw please destroy :)

@cenit Curious as to your thoughts on the opencv patch.

BillyONeal committed:
Add missing =fails.

Neumann-A commented on ports/quirc/CMakeLists.txt:
```diff
@@ -1,17 +1,27 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.19)
```
```suggestion
cmake_minimum_required(VERSION 3.7)
```
is probably more than enough considering the "complexity" of the CMakeLists.txt :)
vcpkg internally uses 3.19 but users might not be using that version (especially on unix)

BillyONeal replied:
Right, but this is being built by vcpkg so presumably the version users use doesn't matter?

Neumann-A commented on ports/quirc/CMakeLists.txt:
```diff
@@ -1,17 +1,27 @@
 
-install(FILES ${PROJECT_SOURCE_DIR}/lib/quirc.h DESTINATION include)
+install(EXPORT quirc-targets
+    FILE quirc-config.cmake
```
in general targets should not be directly exported as -config.cmake but in this case the it is ok since the targets don't depend on any other imported targets.

cenit commented:
fix for opencv4 looks ok. Should be performed identical on opencv3. I can do it after this PR is merged inside #15162

cenit commented:
fix for opencv4 looks ok but might require an additional find_dependency in the OpenCVConfig.cmake file.
Regressions look like it is necessary :)

In any case, a similar fix should be applied also on opencv3. I can do it after this PR is merged inside #15162

cenit commented:
tesseract failures are not there in pr, only in ci: they are due to a Params.h (going by memory) header which is found in the installed/(triplet)/include folder, which clash with tesseract's Params.h file and is found first.
It might require a patch in tesseract to first look into its own buildtrees folder and then inside dependencies include folders (which means also the triplet root include folder)

cenit commented:
tesseract failures are not there in pr, only in ci: they are due to a Params.h (going by memory, I might be wrong about the file name) header which is found in the installed/(triplet)/include folder, which clashes with tesseract's Params.h file and is found first.
It might require a patch in tesseract to first look into its own buildtrees folder and then inside dependencies include folders (which means also the triplet root include folder). Otherwise (but it might be more difficult and less correct) we could find out the port which installs Params.h (it's not a dependency of tesseract, it's a random one) and install it in an include subfolder

cenit commented:
> Probably broken by #15089, fix here

not broken by, just uncovered, because finally opencv was re-included in the tested ports! It was considered a cascade failure for a long time. I didn't even noticed it and that's a problem, because for a long time anything that broke opencv passed under the radar (and was finally fixed in that PR).
In fact, one thing that I was asking you in PR #15162 was how to better deal with basic ports failure. If you baseline any failure from any of them, you are implicitly removing from testing going onwards all depending ports, without any notice to the PR author and reviewers

BillyONeal commented:
>In fact, one thing that I was asking you in PR #15162 was how to better deal with "basic" (fundamental) ports failure. If you baseline any failure from any of them, you are implicitly removing from testing going onwards all ports depending on it, without any notice to the PR author and reviewers

I think the right fix long term is that cascade needs to be equivalent to fail for purposes of `supports:` and `ci.baseline.txt.`

BillyONeal commented:
>In any case, a similar fix should be applied also on opencv3. I can do it after this PR is merged inside #15162

Ah, I missed that since it's disabled in CI :). I thought the quirc thing must have been new.

cenit commented:
> >In fact, one thing that I was asking you in PR #15162 was how to better deal with "basic" (fundamental) ports failure. If you baseline any failure from any of them, you are implicitly removing from testing going onwards all ports depending on it, without any notice to the PR author and reviewers
> 
> 
> 
> I think the right fix long term is that cascade needs to be equivalent to fail for purposes of `supports:` and `ci.baseline.txt.`

Which are very different in concept, if I understand correctly: if you declare a platform not supported is not even tested, if you add to ci.baseline.txt otoh it is tested and might appear as "passing, remove from failing list" in case a fix somewhere gives that port the green light too.

ras0219 commented on ports/opencv4/vcpkg.json:
```diff
@@ -0,0 +1,216 @@
+  "description": "computer vision library",
+  "homepage": "https://github.com/opencv/opencv",
+  "dependencies": [
+    "quirc",
```
I think an off-by-default feature would be better here.

ras0219 replied:
+this probably needs a `find_dependency()` call like @cenit mentioned.

BillyONeal replied:
That seems counter to what OpenCV's authors intend?

> OCV_OPTION(WITH_QUIRC "Include library QR-code decoding" **ON**
>   VISIBLE_IF TRUE
>   VERIFY HAVE_QUIRC)

?

BillyONeal replied:
(To be clear, doing a feature makes sense; I'm quibbling over the default)

cenit replied:
do a feature and put it as a default feature. That would be the best IMHO

ras0219 replied:
On by default also seems reasonable since opencv's default when external QUIRC is missing is to build the vendored version. If the behavior upon missing the external dep would be to disable the feature, I think it would best to have it off-by-default.

BillyONeal commented:
> fix for opencv4 looks ok but might require an additional find_dependency in the OpenCVConfig.cmake file.
> Regressions look like it is necessary :)

Do you have suggestions? I don't see any obvious place to add find_dependency in here :(

cenit commented:
> > fix for opencv4 looks ok but might require an additional find_dependency in the OpenCVConfig.cmake file.
> > Regressions look like it is necessary :)
> 
> Do you have suggestions? I don't see any obvious place to add find_dependency in here :( @cenit

https://github.com/microsoft/vcpkg/blob/3b433e5081f35a32331492d98a8b0c1c2477048e/ports/opencv4/portfile.cmake#L430-L432
^here , something similar to other dependencies

nardl commented on ports/opencv4/0011-devendor-quirc.patch:
```diff
@@ -0,0 +1,26 @@
+@@ -2,7 +2,5 @@ set(the_description "Object Detection")
+ ocv_define_module(objdetect opencv_core opencv_imgproc opencv_calib3d WRAP java python js)
+ 
+ if(HAVE_QUIRC)
```
Hurensohn

nardl reviewed (approved):
Hurnesojn

nardl reviewed (approved):
Hurnesojn

nardl reviewed (approved):
Hurnesojn

cenit commented:
I did not integrate quirc fixes, but I did fully extract the port from opencv in #15859 (complete with downstream dependency fixes).
Please let me know how to go on with both these PR. Shall I integrate quirc fixes also there?

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into build_fixes

# Conflicts:
#	ports/opencv4/CONTROL
#	versions/baseline.json
#	versions/o-/opencv4.json

BillyONeal committed:
Update opencv4 devendor patch for 4.5

BillyONeal committed:
Transform quirc into a feature.

BillyONeal committed:
Repeat changes for opencv3.

BillyONeal committed:
Repeat for opencv. Also fix version for "opencv".

BillyONeal commented:
The string "quirc" does not appear in opencv2 so I'm assuming it doesn't need edits.

BillyONeal commented:
> ~I did not integrate quirc fixes, but I did fully extract the port from opencv in #15859 (complete with downstream dependency fixes).~
> I copied and completed quirc fixes also in #15859 in order to properly check that PR.
> Please let me know how to go on with both these PR.

I would prefer to merge this before that since this is a build fix and that is a version change, but it isn't a strong opinion

BillyONeal commented:
Upcoming dxut change has no source changes:
![image](https://user-images.githubusercontent.com/1544943/105891053-07439e00-5fc5-11eb-81e4-cbc9c5b649ce.png)

BillyONeal committed:
[dxut] Use vcpkg_from_github, correct "supports",

if(NOT VCPKG_CRT_LINKAGE STREQUAL "dynamic")
  message(FATAL_ERROR "DXUT only supports dynamic CRT linkage")
endif()

BillyONeal committed:
[epsilon] Skip in ci.baseline.txt for now because it has "vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)" but also checks against the CRT linkage making the right fix unclear.

BillyONeal committed:
Fix flipped supports:

strega-nil commented on ports/opencv3/portfile.cmake:
```diff
@@ -401,6 +403,9 @@ find_dependency(Tesseract)")
   if(BUILD_opencv_ovis)
     string(APPEND DEPS_STRING "\nfind_dependency(Ogre)\nfind_dependency(Freetype)")
   endif()
+  if("quirc" IN_LIAST FEATURES)
```
```suggestion
  if("quirc" IN_LIST FEATURES)
```

strega-nil reviewed (approved):
Just the one thing.

strega-nil committed:
Update ports/opencv3/portfile.cmake

BillyONeal committed:
x-add-version
